File: codes/javascript/polymorphism/es6/overriding.js
#!/usr/bin/env node

class Person {
    talk(){ console.log("I am a person"); }
}

class Student extends Person {
    talk(){ console.log("I am a student"); }
}

let pe = new Person(),
    st = new Student();
pe.talk(); // I am a person
st.talk(); // I am a student

File: codes/perl/polymorphism/overriding.pl
package Person;
sub new { return bless {}, shift; }
sub talk { print "I am a person\n"; }
1;

package Student;
our @ISA = qw(Person);
sub talk { print "I am a student\n"; }
1;

$pe = Person->new();
$st = Student->new();
$pe->talk(); # I am a person
$st->talk(); # I am a student
